sdk/database/helper/credsutil/sql.go

```diff
@@ -19,6 +20,14 @@ type SQLCredentialsProducer struct {
 	Separator      string
 }
 
+func (scp *SQLCredentialsProducer) GenerateCredentials(ctx context.Context) (string, error) {
```

Comment:
Is `GenerateCredentials` functionally the same as `GeneratePassword` but with a context provided? Why do we need to wrap `GeneratePassword` around this?

Reply:
`GenerateCredentials` game late in the process and maybe bad design resulting from tunnel vision. We needed a way to get the plugin to generate a password for us to use, so we knew it, before we told the plugin to set the password to that value. In things like `CreateUser` the password generation happens plugin-side and the server side didn't care about knowing it before hand, but for storing that pw in WAL entries we needed to know it first.

It is essentially just a wrapper around `GeneratePassword`, although that method is not actually part of the dbplugin interface to be a database plugin. It has the context because it's called over gRPC.